src/persist/src/file.rs

```diff
@@ -322,16 +323,27 @@ impl Blob for FileBlob {
 
     fn set(&mut self, key: &str, value: Vec<u8>, allow_overwrite: bool) -> Result<(), Error> {
         let file_path = self.blob_path(key)?;
-        let mut file = if allow_overwrite {
-            File::create(file_path)?
+        if allow_overwrite {
+            // allow_overwrite=true requires atomic writes, so write to a temp
+            // file and rename it into place.
+            let mut tmp_name = file_path.clone();
+            debug_assert_eq!(tmp_name.extension(), None);
+            tmp_name.set_extension(UNIX_EPOCH.elapsed().unwrap().as_nanos().to_string());
+            let mut file = OpenOptions::new()
+                .write(true)
+                .create_new(true)
+                .open(&tmp_name)?;
+            file.write_all(&value[..])?;
+            file.sync_all()?;
+            fs::rename(tmp_name, &file_path)?;
```

Comment:
i _think_ you have to also fsync the parent directory to persist the rename correctly (at least on some file systems). that being said, you probably also have to do that on file create time. I'm ok leaving it as a todo / to figure out later for now.

Reply:
yeah you're right. I don't see anything in the stdlib for this, so added the TODO for now